storage_image_product/models/image_tag.py

```diff
@@ -3,13 +3,25 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ImageTag(models.Model):
     _name = "image.tag"
 
+    @api.model
+    def _get_default_apply_on(self):
+        active_model = self.env.context.get("active_model")
+        return (
+            "product"
+            if active_model == "product.image.relation"
+            else "category"
+            if active_model == "category.image.relation"
+            else False
+        )
+
     name = fields.Char()
     apply_on = fields.Selection(
-        selection=[("product", "Product"), ("category", "Category")]
+        selection=[("product", "Product"), ("category", "Category")],
+        default=_get_default_apply_on,
```

Comment:
To ensure to make it overridable, shouldn't we use lambda function ?

Reply:
Indeed!